pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workloadrebalancer
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	appsv1alpha1 "github.com/karmada-io/karmada/pkg/apis/apps/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+const (
+	// ControllerName is the controller name that will be used when reporting events.
+	ControllerName = "workload-rebalancer"
+)
+
+// RebalancerController is to handle a rebalance to workloads selected by WorkloadRebalancer object.
+type RebalancerController struct {
+	Client client.Client
+}
+
+var predicateFunc = predicate.Funcs{
```

Comment:
Wouldn't it be better to make it a local variable of the `SetupWithManager` method?

Reply:
I considered two points:
1. put it into `SetupWithManager` doesn't save much performance overhead.
2. it may become complicated in the future and it will be poorly readable when placed in functions.

Reply:
Our other controllers are actually written consistently, and it's probably always good to be consistent. I understand that there is no difference in readability before and after.